agent/hcp/telemetry/otel_sink.go

```diff
@@ -38,7 +42,7 @@ type OTELSink struct {
 	// spaceReplacer cleans the flattened key by removing any spaces.
 	spaceReplacer *strings.Replacer
 	logger        hclog.Logger
```

Comment:
This is fine but nice to not have to have loggers on structs. Unfortunate that go-metrics doesn't pass a context.